### PR TITLE
Client condition support

### PR DESCRIPTION
**Pull request by blide**
It replaces ClientIDs list to ClientCondition function that validates token as you want.

### CODE REVIEW

**Comment by emanoelxavier** [comment]
Thanks for sending the PR. Can you please created an issue describing the requirement/scenario/bug. That will help with the code review and merge.

**Comment by blide** [comment]
Sometimes you need to keep client ids in db or even to give access to any logged user. Patch covers those usecases.

**Comment by emanoelxavier** [comment]
The first requirement, fetching clients from the db, can already be met within your own implementation of the provider getter. See the example getProviders_googleplayground but instead of hard coding the ids you can fetch them from the db or a configuration. The second requirement of completely skipping the audience validation makes sense mostly only for test scenarios, per openid recommendation the audience should always be validated. 
Also a more consistent way of achieving that might be to have a new extension point passed all the way down to the tokenvalidator, that extension point would take place of the validateAudiences so the user could implement his own method that always returns true and pass it in the middleware Configuration as another option, if not provided then we would default to the internal implementation of the validateAudiences. 
If you feel strongly about this second requirement (skipping audience validation for test scenarios) you can try out this implementation and send another PR or create an issue and I can tackle it.

File: openid/idtokenvalidator.go [hunk 1]
```diff
@@ -140,25 +140,23 @@ func validateAudiences(jt *jwt.Token, p *Provider) (string, error) {
 		return "", err
 	}
 
-	for _, aud := range p.ClientIDs {
-		for _, audienceClaim := range audiencesClaim {
-			ta, ok := audienceClaim.(string)
-			if !ok {
-				fmt.Printf("aud type %T \n", audienceClaim)
-				return "", &ValidationError{Code: ValidationErrorInvalidAudienceType, Message: fmt.Sprintf("Invalid Audiences type: %T", audiencesClaim), HTTPStatus: http.StatusUnauthorized}
```
Comment (emanoelxavier) [inline openid/idtokenvalidator.go]: For the requirement of skipping the audiences validation altogether we should just allow the user to provide his own implementation of the validateAudiences function. If not provided then we can fall back to the internal validatedAudiences without changes. That extension point should be registered in the middleware configuration as another option.

**Comment by blide** [comment]
Well. I got it. For the first requirement the point is that now you fetch clients once at start up, but it sounds weird if you need to update clients during.

**Comment by emanoelxavier** [comment]
Note that while the ProvidersGetter is set at start up it will be invoked when the token is validated for each received request. So if your own implementation of that function fetches that information from a database then new clients added in the added in the db will take effect without the need to restart the application. Also for that reason you should probably cache this information with some reasonable TTL in order to avoid DB round trips for every request.

**Comment by blide** [comment]
Oh.  I didn't note that. Thanks for explanation

**Pull request closed without merging**